lib/rouge/lexers/ecl.rb

```diff
@@ -0,0 +1,136 @@
+        unstable update use width
+      )
+
+      functions = %w(
+        abs acos aggregate allnodes apply ascii asin asstring 
+        atan _token ave case catch choose choosen choosesets 
+        clustersize combine correlation cos cosh count covariance cron 
+        dataset dedup define denormalize dictionary distribute distributed distribution 
+        ebcdic enth error evaluate event eventextra eventname exists 
+        exp failcode failmessage fetch fromunicode fromxml getenv getisvalid 
+        global graph group hash hashcrc having httpcall httpheader 
+        if iff index intformat isvalid iterate join keyunicode 
+        length library limit ln local log loop map 
+        matched matchlength matchposition matchtext matchunicode max merge mergejoin 
+        min nofold nolocal nonempty normalize parse pipe power 
+        preload process project pull random range rank ranked 
+        realformat recordof regexfind regexreplace regroup rejected rollup round 
+        roundup row rowdiff sample set sin sinh sizeof 
+        soapcall sort sorted sqrt stepped stored sum table 
+        tan tanh thisnode topn tounicode toxml transfer transform 
+        trim truncate typeof ungroup unicodeorder variance which workunit 
+        xmldecode xmlencode xmltext xmlunicode apply assert build buildindex 
+        evaluate fail keydiff keypatch loadxml nothor notify output 
+        parallel sequential soapcall wait
+      )
+
+      class1 = %w(
+        file date str math metaphone metaphone3 uni audit blas system
+      )
+
+      class2 = %w(
+        debug email job log thorlib util workunit
+      )
+
+      state :single_quote do
+        rule %r([xDQUV]?'([^'\\]*(?:\\.[^'\\]*)*)'), Str::Single
+        rule %r/\\(x\\h{2}|[0-2][0-7]{,2}|3[0-6][0-7]?|37[0-7]?|[4-7][0-7]?|.|$)/, Text
+      end
+
+      state :inline_whitespace do
+        rule %r/[ \t\r]+/, Text
+        rule %r/\\\n/, Text # line continuation
+        rule %r(/[*].*?[*]/)m, Comment::Multiline
+      end
+
+      state :whitespace do
+        rule %r/\n+/m, Text
+        rule %r(//.*), Comment::Single
+        mixin :inline_whitespace
+      end
+
+      state :root do
+	mixin :whitespace
+	mixin :single_quote
+
+        rule %r(\b(?i:(begin|end)c\+\+)), Keyword
+        rule %r(\b(?i:(#{keywords.join('|')}))\b), Keyword
+        rule %r(#(?i:(#{template.join('|')}))\b), Keyword::Declaration
+        rule %r(\b(?i:(std)\.(#{class1.join('|')})\.(#{class2.join('|')}))\b), Name::Class
+
+        rule %r(\b(?i:(#{typed.join('|')}))\d+\b), Keyword::Type
+        rule %r(\b(?i:(#{type.join('|')}))\b), Keyword::Type
+        rule %r(\b(?i:(integer|unsigned))[1-8]\b), Keyword::Type
+        rule %r(\b(?i:real)(4|8)\b), Keyword::Type
+        rule %r(\b(?i:(u)?decimal)(\d+(_\d+)?)\b), Keyword::Type
+
+        rule %r(\b(?i:(and|not|or|in))\b), Operator::Word
+        rule %r([:=|>|<|<>|/|\\|+|-|=]), Operator 
```

Comment:
I'm not sure I made the point very well in the earlier comment.

You've put the operators inside square brackets meaning that everything in between them is treated as members of a character class. In this case `|` is not treated as matching one or the other, it's matching the literal `|` character. To do what you want here you just need to do this: `[:=></\\+-]` (the `-` goes at the end because it otherwise has a special meaning inside a character class). 

You may want to add a `+` after the closing square bracket so that you can match multiple operators in a single pass (although that's not necessary).

Reply:
@pyrmont we would need `:=` and `<>` to be evaluated as a single operator...

Reply:
@GordonSmith Checking syntax correctness is a non-goal for Rouge and so, from our perspective, it's not a problem if you lex characters as tokens that are invalid syntactically. You could achieve what I think you want by having a regex that's `%r(:=|>|<|...)` but that's inconsistent with how the other lexers are written and is using Rouge more as a parser than a lexer.

Reply:
@pyrmont understood - I think "need" is the wrong term, its more consistency, if the following should not be colored as a comment:
```
/ *  some comments * /
```
Then the same should be true for `:=`, IOW the following should not color it is an operator:
```
xxx :  = yyy;
```
or am I misunderstanding your original remark?

BTW Thanks for your detailed reviews in all of this as we don't have any ruby experience (and I only have minor regex experience).

Finally - do prefer squashed commits during review?

Reply:
Sorry—I haven't finished thinking about this. I was doing some maintenance of other lexers yesterday and noticed they do define set operators so am not so certain of my opinion here.

In terms of squashing commits, definitely prefer that you don't do this! What you've been doing so far has been great; for review purposes it's very helpful to be able to compare individual commits! I'll do a squash when I merge it into master.

Reply:
@pyrmont, Please resolve if this discussion is complete.

Reply:
@dabittan I've changed my mind and think that it's fine to specify precisely the acceptable operators but the way regex pattern that's currently being used isn't the way to do it. Characters in a class (i.e. that are in inside '[' and ']' are matched individually. To achieve what you want, you need to write something like this:

```ruby
rule %r(:=|>|<|<>|/|\\|+|-|=), Operator
```

Reply:
@dehilsterlexis, please let us know when you complete these changes

Reply:
@dabittan @pyrmont completed the changes.

Reply:
Sorry, I forgot to escape the `+`. It should be:

```ruby
rule %r(:=|>|<|<>|/|\\|\+|-|=), Operator
```

Reply:
@pyrmont `rule %r(:=|>|<|<>|/|\\|+|-|=), Operator` does not work. It works with the brackets so I changed it back.

Reply:
Sorry, I sent it a bit later but did you see my comment? The `+` needed to be escaped.